src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -1563,9 +1582,17 @@ ValidatorList::getJson() const
         curr[jss::available] =
             pubCollection.status == PublisherStatus::available;
 
-        auto appendList = [](PublisherList const& publisherList,
-                             Json::Value& target) {
-            target[jss::uri] = publisherList.siteUri;
+        auto appendList = [&context](
+                              PublisherList const& publisherList,
+                              Json::Value& target) {
+            if ((context && context->apiVersion == 1) || !context)
+                target[jss::uri] = publisherList.sourceUri.empty()
+                    ? publisherList.sourcePeer
+                    : publisherList.sourceUri;
+            if (!publisherList.sourceUri.empty())
+                target[jss::source_uri] = publisherList.sourceUri;
```

Comment:
You know, @RichardAH has a good point in his comment https://github.com/ripple/rippled/pull/3984#issuecomment-1144564520. 

What do you think about completely removing the new `jss::source_uri` field, and just including the `sourceUri` member as `jss::uri`? You could get rid of `context` entirely.